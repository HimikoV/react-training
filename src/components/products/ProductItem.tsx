import {
  Card,
  CardContent,
  Box,
  Typography,
  CardMedia,
  IconButton,
} from "@mui/material";
import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
import React from "react";
import { productType } from "../../types/productType";
import ProductSnackBar from "./ProductSnackBar";

const ProductItem = ({
  item,
  addToBasket,
}: {
  item: productType;
  addToBasket: (item: string) => void;
}) => {
  return (
    <Box
      sx={{
        m: 1,
        fontSize: "0.875rem",
        fontWeight: "700",
        width: 230,
      }}
    >
      <Card>
        <CardMedia
          component="img"
          height="250"
          image="https://images.unsplash.com/photo-1515471897120-85416077e011?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8Y29mZmVlJTIwYmFnfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=500&q=60"
          alt="green iguana"
        />
        <CardContent>
          <Box
            sx={{
              display: "flex",
              flexWrap: "wrap",
              color: "#3d405b",
              justifyContent: "space-between",
              alignItems: "flex-start",
              flexDirection: "row",
            }}
          >
            <Box
              sx={{
                display: "flex",
                flexWrap: "wrap",
                color: "#3d405b",
                justifyContent: "flex-start",
                alignItems: "flex-start",
                flexDirection: "column",
              }}
            >
              <Typography
                sx={{ fontSize: 14 }}
                color="text.secondary"
                gutterBottom
              >
                {item.name}
              </Typography>
              <Typography variant="h5" component="div">
                {item.price}$
              </Typography>
            </Box>
            <Box
              sx={{
                display: "flex",
                flexWrap: "nowrap",
                color: "#3d405b",
                justifyContent: "flex-end",
                alignSelf: "flex-end",
                flexDirection: "column",
              }}
            >
              <IconButton
                aria-label="delete"
                onClick={() => addToBasket(item.id)}
              >
                <ShoppingCartIcon />
              </IconButton>

              <ProductSnackBar text="Product added to the basket" />
            </Box>
          </Box>
        </CardContent>
      </Card>
    </Box>
  );
};

export default ProductItem;
