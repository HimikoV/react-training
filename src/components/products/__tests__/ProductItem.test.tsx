import ProductItem from "../ProductItem";
import { describe, it } from "@jest/globals";
import renderer from "react-test-renderer";

describe("testing <ProductItem/> render", () => {
  it("ProductItem Renders Correctly", () => {
    const productPage = renderer
      .create(
        <ProductItem
          item={{
            name: "product",
            price: 20,
            category: "category_1",
            id: "item_1",
          }}
        />
      )
      .toJSON();
    expect(productPage).toMatchSnapshot();
  });
});
