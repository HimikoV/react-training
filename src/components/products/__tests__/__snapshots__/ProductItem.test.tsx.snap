// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <ProductItem/> render ProductItem Renders Correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-6ixihs-MuiPaper-root-MuiCard-root"
>
  <img
    alt="green iguana"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
    height="250"
    src="https://images.unsplash.com/photo-1515471897120-85416077e011?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8Y29mZmVlJTIwYmFnfGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=500&q=60"
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiGrid-root css-i4z4p2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <span
          className="MuiTypography-root MuiTypography-caption css-1d9s20a-MuiTypography-root"
        >
          product
        </span>
        <h5
          className="MuiTypography-root MuiTypography-h5 css-1invrju-MuiTypography-root"
        >
          20
          $
        </h5>
      </div>
      <div
        className="MuiGrid-root css-gqim7x-MuiGrid-root"
      >
        <button
          aria-label="delete"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ShoppingCartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
